File: Lab5/Question7/src/Animal.java
public class Animal extends LivingThing{
    private double weight;

    public Animal(String name, double weight) {
        super(name);
        this.weight = weight;
    }

    @Override
    public String getName() {
        return super.getName();
    }

    public double getWeight() {
        return weight;
    }

    public void setWeight(double weight) {
        this.weight = weight;
    }

    public static void testAnimal() {
        Dog.testDog();
        Cat.testCat();
    }
}

File: Lab5/Animal.kt
open class Animal : LivingThing {
    private var weight: Double

    constructor(name: String, weight: Double) : super(name) {
        this.weight = weight
    }

    override fun getName(): String {
        return super.getName()
    }

    open fun getWeight(): Double {
        return weight
    }

    fun setWeight(weight: Double) {
        this.weight = weight
    }
}